code/game/machinery/camera/camera_assembly.dm

```diff
@@ -105,6 +105,8 @@
 	if(is_type_in_list(W, possible_upgrades) && !is_type_in_list(W, upgrades)) // Is a possible upgrade and isn't in the camera already.
 		if(!user.unEquip(W))
 			return
+		if(!W.drop_item()) // no eating borg items
+			return

```

Comment:
The borg's analyzer does not have NODROP. drop_item() has a special check to fail for robot users, unEquip() doesn't have that.